conseil-common/src/main/scala/tech/cryptonomic/conseil/common/util/CollectionOps.scala

```diff
@@ -27,6 +27,16 @@ object CollectionOps {
     def byKey(): Map[K, Seq[V]] = groupByKey(seq)
   }
 
+  /** allows counting the occurrences of values in
+    * a generic sequence that might contain duplicates
+    *
+    * @param seq the original sequence
+    */
+  implicit class SeqCounter[T](seq: Seq[T]) {
```

Comment:
Because it's super-verbose and noisy on the call site, and in my experience with scala I've found this pattern to show up many times.

It's actually more general, but I didn't generalise here.

The pattern is to count grouped values in a collection, using a projection function. In this case we use identity as projection since we only want to count values based on equality.

But you can count grouping on a specific field, e.g. you have operations and you want to count how many for each kind, or multiple ballots and you want to count by vote.

I think that having a specific name makes it clear on the usage site, much more than this groupBy mapValues

Reply:
Ok, thanks.